src/components/tracked-controls.js

```diff
@@ -35,6 +36,27 @@ module.exports.Component = registerComponent('tracked-controls', {
 
     this.previousControllerPosition = new THREE.Vector3();
     this.updateGamepad();
+    this.onComponentInitialized = bind(this.onComponentInitialized, this);
+  },
+
+  play: function () {
+    this.el.addEventListener('componentinitialized', this.onComponentInitialized);
+  },
+
+  pause: function () {
+    this.el.removeEventListener('componentinitialized', this.onComponentInitialized);
+  },
+
+  onComponentInitialized: function (evt) {
+    if (evt.detail.name !== 'tracked-controls') { return; }
+    var self = this;
+    Object.keys(this.el.components).forEach(function (name) {
+      const component = self.el.components[name];
+      if (component.controllerPresent) {
+        // Emit controllerpresent event.
+        self.el.emit('controllerpresent', {name: name, component: component});
```

Comment:
so you are 90% of the way to `auto-detect-controllers` approach now :-)
where `auto-detect-controllers` was the component stand-in that would listen for new controller available, and inject whatever was appropriate.

A little strange to have `tracked-controls` bottoms-up and `oculus-touch-controls`, `vive-controls` etc. top-down though, rather than having `auto-detect-controllers` that would listen for stage events and inject the higher-level components (oculus, vive) not just the lower-level (tracked-controls)

can we agree on names for the scene events
(closely matching wrapped Gamepad API events)
`controllerconnected`
`controllerdisconnected`
?

I'd suggest that we maintain `controllerpresent` as the name of the event for the higher-level component that actually picks up.